### DISCUSSION TITLE
how to use mapping node with gradient

### DISCUSSION THREAD

**Discussion by stann-co** [bnpr/Malt#543] [Q&A] [answered]
for  extra context, i'm trying to figure out how to recreate some basic shader stuff like in the normal render engines.
Like adding a gradient from the bottom of a mesh to the top. Normally i'd use a gradient and mapping node.
But here that is not as intuitive.

the gradient node takes in just a float. But the mapping node outputs vec3, and i can't find a way to split vec3 into different channels

**Comment by pragma37** [comment, marked as answer]
The `Math > Vector` nodes have `Combine` and `Separate` functions.
Note that you can use the `Node Add > Search... (Malt)`  operator for searching function names too, not just node names.

**Reply by stann-co** [reply]
Okay! I was searching for split and break, and couldn't find it
